resources/etcd_installation_binary.rb

```diff
@@ -30,16 +30,25 @@
     action :create
   end
 
+  directory new_resource.etcd_bin_prefix do
+    action :create
+  end
+
+  file 'etcd_checksum' do
+    path "#{new_resource.etcd_bin_prefix}/.etcd_checksum"
+    content new_resource.checksum
+  end
```

Comment:
We need to store the checksum in a file so that we trigger the extraction of `etcd` and `etcdctl` when the checksum (hence the etcd version) changes.